crates/sui-core/src/checkpoints/mod.rs

```diff
@@ -419,7 +418,6 @@ impl CheckpointStore {
             .is_none());
         debug_assert!(self.next_checkpoint() == checkpoint_sequence_number);
 
-        self.check_checkpoint_transactions(contents.transactions.iter(), &effects_store)?;
```

Comment:
Removing this does have a consequence: now it's possible for `handle_internal_batch` to pass over transactions that are already checkpointed. Currently we print an error whenever that happens. But if this is to become a normal case, we should remove that error.

Reply:
Good point!